app/controllers/application_controller.rb

```diff
@@ -3,11 +3,47 @@ class ApplicationController < ActionController::Base
   # Prevent CSRF attacks by raising an exception.
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
-  helper_method :current_division
 
+  helper_method :selected_division_id
+  helper_method :division_select_options
+
+  # Represents the division to use when creating new entities.
   def current_division
-    # TODO
-    Division.root
+    selected_division || default_division
+  end
+
+  # Division to uses for new entities if a division is not specifically selected
+  def default_division
+    current_user.default_division
+  end
+
+  # Represents the current division filter applied to index views.
+  def selected_division
+    id = selected_division_id
+    if id
+      Division.find_safe(id)
+    else
+      nil
+    end
+  end
+
+  def division_index_filter
```

Comment:
Please add a comment for this method as it's not super clear what it's doing (especially the descendant_ids vs accessible_ids bit).